### PR TITLE
#1664 : Escape RegExp pattern in Java.

### LINKED ISSUE #1664: entity generator - String field RegExp pattern not escaped in Java
Generated an entity with a string field validated with this regular expression pattern:
^(?:[0-9]{1,3}\\.){3}[0-9]{1,3}$
to validate IPv4 addresses.

Compiling the application fails with:
```java
MyEntity.java:32: error: illegal escape character
    @Pattern(regexp = "^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$")
```
(It doesn't like that `\.` in the middle there.

MyEntity.java, should be generated as:
```java
    @NotNull
    @Pattern(regexp = "^(?:[0-9]{1,3}\\.){3}[0-9]{1,3}$")
    @Field("ip")
    private String ip;
```
(With the escaped `\\.`)

The javascript side functions fine. Its a minor issue that is easily corrected after entity generation but does cause compile time error.

JHipster v2.17.0

### PR DESCRIPTION
**Pull request by Falydoor**
Escape RegExp pattern on entity validation in Java

Fix #1664

### CODE REVIEW

**Comment by jdubois** [comment]
Thanks!!

**Comment by jdubois** [comment]
@Falydoor hey, I didn't see you where from Ippon, great! Hope to see you at the office when my current gig is over :-)

**Comment by jdubois** [comment]
Argh, it causes an issue in the entity generator!!

To reproduce, generate a new entity, with a new field, type String, no validation rule. You will get:

    TypeError: Cannot read property 'replace' of undefined
      at EntityGenerator.<anonymous> (/Users/julien/workspace/generator-jhipster/entity/index.js:565:79)

I'm reverting the commit.

File: entity/index.js [hunk 1]
```diff
@@ -562,11 +562,12 @@ EntityGenerator.prototype.askForFields = function askForFields() {
                 fieldValidateRulesMinlength: props.fieldValidateRulesMinlength,
                 fieldValidateRulesMaxlength: props.fieldValidateRulesMaxlength,
                 fieldValidateRulesPattern: props.fieldValidateRulesPattern,
+                fieldValidateRulesPatternJava: props.fieldValidateRulesPattern.replace(/\\/g, '\\\\'),
```
Comment (andidev) [inline entity/index.js]: props.fieldValidateRulesPattern is not always defined so this line result that no other validation rule works without the regex one.

see stacktrace
```bash
$ jhe Oke
The entity Oke is being created.
Generating field #1
? Do you want to add a field to your entity? Yes
? What is the name of your field? lala
? What is the type of your field? String
? Do you want to add validation rules to your field? Yes
? Which validation rules do you want to add? Maximum length
? What is the maximum length of your field? 100
/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:579
    throw e;
          ^
TypeError: Cannot read property 'replace' of undefined
    at EntityGenerator.<anonymous> (/Users/anders/Workspace/generator-jhipster/entity/index.js:565:79)
    at EntityGenerator.<anonymous> (/Users/anders/Workspace/generator-jhipster/node_modules/yeoman-generator/lib/base.js:233:7)
    at PromptUI.onCompletion (/usr/local/lib/node_modules/yo/node_modules/inquirer/lib/ui/prompt.js:69:10)
    at AnonymousObserver.Rx.AnonymousObserver.AnonymousObserver.completed (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:1977:12)
    at AnonymousObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:1914:14)
    at AnonymousObserver.tryCatcher (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:567:29)
    at AutoDetachObserverPrototype.completed (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:5032:56)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:1914:14)
    at InnerObserver.onCompleted (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:3345:65)
    at InnerObserver.tryCatcher (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:567:29)
```

Comment (jdubois) [inline entity/index.js]: Yes I just saw it and reverted the commit! Shouldn't be hard to fix

**Comment by Falydoor** [comment]
Yes sorry totally newbie mistake.

I did a new PR as I did not know well how to continue on this one.

**Pull request merged**